ports/raspberrypi/common-hal/analogio/AnalogFastIn.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * SPDX-FileCopyrightText: Copyright (c) 2022 Lee Atkinson, MeanStride Technology, Inc.
+ * Taken from AnalogIn by Scott Shawcroft for Adafruit Industries
+ * Also from DMA_Capture by Luke Wren of Raspberry Pi (Trading) Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "common-hal/analogio/AnalogFastIn.h"
+#include "shared-bindings/analogio/AnalogFastIn.h"
+#include "shared-bindings/microcontroller/Pin.h"
+#include "py/runtime.h"
+#include "supervisor/shared/translate/translate.h"
+#include "src/rp2_common/hardware_adc/include/hardware/adc.h"
+#include "src/rp2_common/hardware_dma/include/hardware/dma.h"
+//     /sdk/src/rp2_common/hardware_dma/include/hardware/dma.h
+//     ports/raspberrypi/
+#include "sdk/src/common/pico_stdlib/include/pico/stdlib.h"
+
+#define ADC_FIRST_PIN_NUMBER 26
+#define ADC_PIN_COUNT 4
+
+void common_hal_analogio_analogfastin_construct(analogio_analogfastin_obj_t *self, const mcu_pin_obj_t *pin, uint8_t *buffer, uint32_t len, uint8_t bytes_per_sample, bool samples_signed, mp_float_t sample_rate) {
+
+    // Set pin and channel
+    self->pin = pin;
+    claim_pin(pin);
+    self->chan = pin->number - ADC_FIRST_PIN_NUMBER;
```

Comment:
Need to validate that `pin-number` works. If `pin->number` is less than `ADC_FIRST_PIN_NUMBER`, `self->chan` will get a negtive number.